QEfficient/generation/text_generation_inference.py

```diff
@@ -320,44 +209,48 @@ def cloud_ai_100_exec_kv(
     stream: bool = True,
     write_io_dir: Optional[str] = None,
     automation=False,
+    full_batch_size: Optional[int] = None,
 ):
     batch_size, ctx_len = get_compilation_dims(qpc_path)
     prompt: List[str] = get_input_prompts(prompt, prompts_txt_file_path)
-    prompt = fix_prompts(prompt, batch_size)
-
-    prefill_time = []
-    decode_perf = []
-    total_perf = []
-    total_time = []
-    generated_texts = []
-    generated_ids = []
-
-    for i in range(0, len(prompt), batch_size):
-        print("Inference iteration =", i // batch_size)
-        execinfo = cloud_ai_100_exec_kv_helper(
-            tokenizer=tokenizer,
-            prompt=prompt[i : i + batch_size],
-            qpc_path=qpc_path,
-            device_id=device_id,
-            ctx_len=ctx_len,
-            generation_len=generation_len,
-            enable_debug_logs=enable_debug_logs,
-            stream=stream,
-            write_io_dir=write_io_dir,
-        )
-        generated_ids.append(execinfo.generated_ids)
-        generated_texts.append(execinfo.generated_texts)
-        prefill_time.append(execinfo.prefill_time)
-        decode_perf.append(execinfo.decode_perf)
-        total_perf.append(execinfo.total_perf)
-        total_time.append(execinfo.total_time)
-
-    prefill_time = np.average(prefill_time)
-    decode_perf = np.average(decode_perf)
-    total_perf = np.average(total_perf)
-    total_time = np.average(total_time)
-
-    execinfo = CloudAI100ExecInfo(
+    prompt = fix_prompts(prompt, batch_size, full_batch_size)
+    generate_text = TextGeneration(
+        tokenizer=tokenizer,
+        prompt=prompt,
+        qpc_path=qpc_path,
+        device_id=device_id,
+        ctx_len=ctx_len,
+        generation_len=generation_len,
+        enable_debug_logs=enable_debug_logs,
+        stream=stream,
+        write_io_dir=write_io_dir,
+        full_batch_size=full_batch_size,
+    )
+    if full_batch_size is None:
+        exec_info = [
+            generate_text.cloud_ai_100_exec_kv_helper(prompt[i : i + batch_size], generation_len)
+            for i in range(0, len(prompt), batch_size)
+        ]
+        prefill_time = np.average([info.prefill_time for info in exec_info])
+        decode_perf = np.average([info.decode_perf for info in exec_info])
+        total_perf = np.average([info.total_perf for info in exec_info])
+        total_time = np.average([info.total_time for info in exec_info])
+        generated_texts = [info.generated_texts for info in exec_info]
+        generated_ids = [info.generated_ids for info in exec_info]
+    else:
+        exec_info = generate_text.cloud_ai_100_exec_kv_helper(prompt=prompt, generation_len=generation_len)
+        # prefill_time = np.average(exec_info.prefill_time)
```

Comment:
We dont need to average in CB because we are processing all inputs in one go. In regular execution if the input length exceeds the BS, we loop through the inputs. But in the case of CB the latency we capture is for the entire set of inputs.